### DISCUSSION TITLE
Error Maven - mvn test me llena todo de errores que no entiendo

### DISCUSSION THREAD

**Discussion by ivanscardaoni1** [dds-jv/foro#180] [Consultas de herramientas] [answered]
Envio las fotos de lo que me tira cuando corro: mvn clean, luego mvn compile, y por ultimo mvn test, donde me saltan una serie de errores que no logro comprender.
<img width="1440" alt="Screen Shot 2021-04-19 at 10 22 56" src="https://user-images.githubusercontent.com/62474581/115244284-2cded000-a0fa-11eb-87b1-90f6da72b38d.png">
<img width="1440" alt="Screen Shot 2021-04-19 at 10 23 11" src="https://user-images.githubusercontent.com/62474581/115244309-323c1a80-a0fa-11eb-8dc0-71dbd37301bb.png">
<img width="1440" alt="Screen Shot 2021-04-19 at 10 23 28" src="https://user-images.githubusercontent.com/62474581/115244326-36683800-a0fa-11eb-9209-0c257a2f17da.png">
<img width="1440" alt="Screen Shot 2021-04-19 at 10 23 53" src="https://user-images.githubusercontent.com/62474581/115244345-3b2cec00-a0fa-11eb-91dd-55bcad39549e.png">
<img width="1440" alt="Screen Shot 2021-04-19 at 10 24 12" src="https://user-images.githubusercontent.com/62474581/115244375-4253fa00-a0fa-11eb-9344-b741c9bdf720.png">

Perdonen por lo extenso, quiero que tengan la mayor info posible para ver de que trata, jaja. 

Saludos.

**Comment by gastonprieto** [comment]
Buenas @ivanscardaoni1 

Que versión de JDK tenes ?

**Reply by ivanscardaoni1** [reply]
Buenas! antes de instalar maven tenia la version 1.8, ahora me aparece 15.0, se ve que algo toque!
<img width="567" alt="Screen Shot 2021-04-19 at 20 25 08" src="https://user-images.githubusercontent.com/62474581/115315277-59700780-a14d-11eb-8822-9a4996ebaf81.png">

Como lo desinstalaria o volveria a la 8?

**Reply by gastonprieto** [reply]
La verdad en mac jamás lo hice, me imagino que es cambiar la `JAVA_HOME`, te fijaste algún tutorial de mac para cambiar de versión de Java ?

**Reply by gabycosta04** [reply]
Hola buenas noches... yo estaria teniendo el mismo problema que ivan pero en windows, pero a diferencia de ivan yo si tengo la version de jdk 1.8
![image](https://user-images.githubusercontent.com/62992579/116647768-bcce1680-a951-11eb-9c7b-9439a74f17c7.png)

**Reply by gabycosta04** [reply]
Que puede ser que tenga algo mal configurado?

**Reply by gastonprieto** [reply]
hola @gabycosta04 podras subirnos el error que te aparece?

**Reply by gabycosta04** [reply]
![image](https://user-images.githubusercontent.com/62992579/116701560-6cc97100-a99e-11eb-9d38-541d15c296a2.png)
![image](https://user-images.githubusercontent.com/62992579/116701589-77840600-a99e-11eb-881c-9809f77540f9.png)
![image](https://user-images.githubusercontent.com/62992579/116701621-8074d780-a99e-11eb-9977-d9f5105ff4eb.png)
ahi esas cosas me aparecen, q por lo que veo son iguales a las de ivan, pero es raro xq me figura que tengo la version 1.8 de java

**Reply by delficostanzo** [reply]
Hola! Me esta pasando lo mismo que Gabriel. Tengo Windows 10 como el y me tira el mismo error pero en mi caso tengo esta version de jdk. 
<img width="431" alt="Captura de pantalla 2021-04-30 221448" src="https://user-images.githubusercontent.com/38230745/116766563-84861100-aa01-11eb-93d8-ed7aad52e2e2.png">

**Reply by nanderson87** [reply]
@delficostanzo 
La versión que temes que tener instalada y/o configurada es la versión 8.

**Reply by delficostanzo** [reply]
@nanderson87 Sii, ahi ya la instale pero una vez que hago "mvn verify" me tira este error:
<img width="917" alt="Captura de pantalla 2021-05-01 110500" src="https://user-images.githubusercontent.com/38230745/116784893-170fca00-aa6d-11eb-892f-8d100b38d0e8.png">
<img width="923" alt="Captura de pantalla 2021-05-01 110446" src="https://user-images.githubusercontent.com/38230745/116784898-19722400-aa6d-11eb-8977-c9ecf80cbac5.png">
<img width="838" alt="Captura de pantalla 2021-05-01 110430" src="https://user-images.githubusercontent.com/38230745/116784900-1aa35100-aa6d-11eb-8932-114761bb9aca.png">
<img width="926" alt="Captura de pantalla 2021-05-01 110410" src="https://user-images.githubusercontent.com/38230745/116784904-1c6d1480-aa6d-11eb-9a36-db942eb59938.png">

Y me fije tambien en ese error que tira que dice: " Medium: Unread field: ar.edu.utn.frba.dds.macowins.Venta.fechaVenta [ar.edu.utn.frba.dds.macowins.Venta] At Venta.java:[line 15] URF_UNREAD_FIELD" que es la siguiente linea:
<img width="292" alt="Captura de pantalla 2021-05-01 110655" src="https://user-images.githubusercontent.com/38230745/116784947-5c33fc00-aa6d-11eb-9fe2-b0984eadece4.png">
y cae justo sobre el constructor. Se que a la fecha la puedo tomar como el tipo Date pero lo considere como un String porque me resultaba mas comodo para testear.

Perdon que mando tantas imagenes, quiero que se vea la mas info posible para ver de que se trata jajaja

Saludos!

**Reply by delficostanzo** [reply]
Ahi lo pude solucionar! Me faltaban agregar mas tests :)
Gracias!

**Comment by LucasArias** [comment]
**_Hola Ivan,_**

Tenia el mismo problema que vos y me tomo varios días hasta poder encontrar la solución. Me volví loco pero en si el problema es muuuuuy simple y reside en que Maven defaultea a la 15. 

![image](https://user-images.githubusercontent.com/59299359/116118631-e5c18380-a693-11eb-8cce-25edfc7f1174.png)

Cambiando la `JAVA_HOME` se soluciona. Te dejo los pasos a seguir, esta super detallados!

![image](https://user-images.githubusercontent.com/59299359/116118464-b4e14e80-a693-11eb-9b39-8d12f3f16663.png)

Te deberían quedar así:

![image](https://user-images.githubusercontent.com/59299359/116118788-0db0e700-a694-11eb-9ac9-0dfa23539ead.png)

_**Espero que te sirva!
Saludos**_

**Reply by gastonprieto** [reply]
te sirvio esto @ivanscardaoni1 ?

**Reply by ivanscardaoni1** [reply]
buenas! gracias a ambos, ahí lo probé y la version de java me sigue dando la 15 no se por que razón. Busque un tutorial pero me lleva a lo mismo (cambiar la JAVA_HOME). Estoy seguro que va por ahí pero algo no estoy pegando.

**Comment by ivanscardaoni1** [comment, marked as answer]
Pude solucionarlo! había instalado java a través de "brew Install java" que te instala la ultima version y te tapa lo que instalaste a través de la web. Hice "brew uninstall java" (sin tener en cuenta las dependencias) eso volvió a la version de java que tenia instalada sin brew, reacomode JAVA_HOME y listo. Gracias a ambos por la ayuda!

**Comment by ivanscardaoni1** [comment]
Otros errores para los que necesito ayuda:

1) Problema cuando le doy al build con el martillito (en algun momento no me lo hacia, y luego de agregar ciertos tests, me lo hizo, tests que aunque comente sigue haciendo lo mismo)

![image](https://user-images.githubusercontent.com/62474581/116721017-39451180-a9b3-11eb-8af7-58bdae1de838.jpeg)


2) mvn verify o mvn test (cualquiera me causa el mismo error)
![image](https://user-images.githubusercontent.com/62474581/116721060-45c96a00-a9b3-11eb-9c57-2d9fb77fbab7.png)
![image](https://user-images.githubusercontent.com/62474581/116721072-48c45a80-a9b3-11eb-9ca4-e493343688dc.png)

Busque en google varias fuentes y pregunte en los grupos, lo que hice por ahora fue:
![image](https://user-images.githubusercontent.com/62474581/116721117-57127680-a9b3-11eb-9d7c-a212a5fb7795.png)
![image](https://user-images.githubusercontent.com/62474581/116721134-5c6fc100-a9b3-11eb-95ac-b38dd78ff762.png)
setear esas dos cosas que decia en google.

Mando lo que me tira la consola cuando le hago 
- "java -version"
- "javac -version"
- "mvn -version"
![image](https://user-images.githubusercontent.com/62474581/116721201-6e516400-a9b3-11eb-8ce5-7b478b720c65.png)

**Reply by nanderson87** [reply]
@ivanscardaoni1

Te pido que me pases algunas cosas:
1. Un print completo del resultado del comando por terminal  `/usr/libexec/java_home -V `
2. Un print completo del comando echo $JAVA_HOME
3. Si un print de como configurantes lo de JAVA_HOME en el archivo `.zshenv`

**Reply by ivanscardaoni1** [reply]
Hola! acá van las fotos.

1) 
<img width="572" alt="Screen Shot 2021-05-01 at 11 35 25" src="https://user-images.githubusercontent.com/62474581/116785822-f007c700-aa71-11eb-93c3-bf40cf8d5453.png">

2) 
<img width="572" alt="Screen Shot 2021-05-01 at 11 35 58" src="https://user-images.githubusercontent.com/62474581/116785828-fa29c580-aa71-11eb-963b-6bed4a8872fa.png">

3)
<img width="571" alt="Screen Shot 2021-05-01 at 11 37 12" src="https://user-images.githubusercontent.com/62474581/116785835-031a9700-aa72-11eb-86c5-614eb9f6d01c.png">

**Reply by nanderson87** [reply]
@ivanscardaoni1, proba lo siguiente:

1. Módifica el valor de JAVA_HOME por /Library/Java/JavaVirtualMachines/jdk1.8.0_291.jdk/Contents/Home.

`export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_291.jdk/Contents/Home`

2. Reinicia la terminal. El comando `java -version` te debería devolver lo mismo y el `echo $JAVA_HOME` este path. Si eso este así, proba algun comando de maven.

A todo esto vos tenes la versión de mac de Big Sur, no?

**Comment by belencaponelli** [comment]
Hola! Estoy teniendo el mismo problema. Tanto el build como los run Test me corren bien, sin ningun error. Sin embargo cuando hago _mvn test_  me falla. 
Version de java: 15.0.2
Version de Maven 3.8.1

Dejo las capturas de los mensajes de error. 

![image](https://user-images.githubusercontent.com/82619083/116765022-99f73d00-a9f9-11eb-805b-a66878faff9c.png)
![image](https://user-images.githubusercontent.com/82619083/116765030-a380a500-a9f9-11eb-905d-93610f9864ca.png)
![image](https://user-images.githubusercontent.com/82619083/116765042-ada2a380-a9f9-11eb-96c9-0ad2a1c839ea.png)


Mil gracias!

**Reply by nanderson87** [reply]
@belencaponelli 

Tenes Windows no? Tenes instalada la versión de JDK 8? Si vas a `C:\Programas Files\Java ` te aparece una carpeta de 1.8 o solo la carpeta jkd-15.0.2? 

Si es la 15, la versión de JDK que necesitas es la 1.8 (o tmb llamada 8)

**Reply by gabycosta04** [reply]
Hola nicolas, una pregunta, tendrias el link para descargar la version JDK 8? porque no la estaria encontrando y yo tambien deberia descargarlo

**Reply by belencaponelli** [reply]
Ahi pude resolverlo, era un tema de las variables de entorno de maven.

Gracias!

**Reply by nanderson87** [reply]
@gabycosta04:

Te dejo el link: https://www.java.com/es/download/ie_manual.jsp
